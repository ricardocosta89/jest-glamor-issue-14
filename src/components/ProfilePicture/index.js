import React from 'react'
import { css } from 'glamor'
//import picturePath from '../../assets/profile.jpg'

const picturePath = '../../assets/profile.jpg'

const ProfilePicture = ({ path = picturePath, alt = 'Profile Picture', side = 140 }) => {
  const pictureSide = `${side}px`
  const pictureStyle = css({
    width: pictureSide,
    height: pictureSide,
    borderRadius: '50%',
    borderWidth: side < 100 ? '2px' : '4px',
    borderColor: '#ffffff',
    borderStyle: 'solid'
  })

  return (
    <img {...pictureStyle} src={path} alt={alt} />
  )
}

export default ProfilePicture