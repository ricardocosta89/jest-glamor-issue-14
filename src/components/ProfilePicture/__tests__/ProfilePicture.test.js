import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ProfilePicture from '../../ProfilePicture'

import { matcher, serializer } from 'jest-glamor-react'
expect.addSnapshotSerializer(serializer)
expect.extend(matcher)

describe('<ProfilePicture />', () => {
  let wrapper

  describe('with default props', () => {
    beforeEach(() => {
      wrapper = shallow(
        <ProfilePicture />
      )
    })

    test('renders the component', () => {
      expect(toJson(wrapper)).toMatchSnapshotWithGlamor(`enzyme.shallow`)
    })
  })
})
