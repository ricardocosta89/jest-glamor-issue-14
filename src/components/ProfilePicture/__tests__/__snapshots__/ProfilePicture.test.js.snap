// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme.shallow 1`] = `
<img
  alt="Profile Picture"
  data-css-iqk43n=""
  src="../../assets/profile.jpg"
/>
`;
